CONTRIBUTING.rst

```diff
@@ -28,88 +50,87 @@ Rasterio's API is different from GDAL's API and this is intentional.
   inputs. This is challenging in practice because GDAL embraces global
   variables.
 
+Git Conventions
+===============
+
+We use a variant of centralized workflow described in the `Git Book
+<https://git-scm.com/book/en/v2/Distributed-Git-Distributed-Workflows>`__.  We
+have no 1.0 release for Rasterio yet and we are tagging and releasing from the
+master branch. Our post-1.0 workflow is to be decided.
+
+Work on features in a new branch of the mapbox/rasterio repo or in a branch on
+a fork. Create a `GitHub pull request
+<https://help.github.com/articles/using-pull-requests/>`__ when the changes are
+ready for review.  We recommend creating a pull request as early as possible
+to give other developers a heads up and to provide an opportunity for valuable
+early feedback.
+
 Code Conventions
 ================
 
 The ``rasterio`` namespace contains both Python and C extension modules. All
 C extension modules are written using `Cython <http://cython.org/>`__. The
 Cython language is a superset of Python. Cython files end with ``.pyx`` and
-``.pxd`` and are where we keep all the code that calls C functions in the GDAL
-library.
+``.pxd`` and are where we keep all the code that calls GDAL's C functions.
 
-Rasterio supports Python 2 and Python 3 in the same code base. We have a
-compatibility module named ``five.py``. Renaming it is a to-do.
+Rasterio supports Python 2 and Python 3 in the same code base, which is
+aided by an internal compatibility module named ``five.py``. Renaming it is a
+to-do (`#719 <https://github.com/mapbox/rasterio/issues/719>`__).  It functions
+similarly to the more widely known `six <https://pythonhosted.org/six/>`__ but
+we only use a small portion of the features so it eliminates a dependency.
 
 We strongly prefer code adhering to `PEP8
 <https://www.python.org/dev/peps/pep-0008/>`__.
 
-Tests are mandatory for new features. We use ``pytest``. Tests written using
-``unittest`` are fine, too.
-
-We aspire to 100% coverage for Python modules. Coverage of the Cython code is
-a future aspiration.
+Tests are mandatory for new features. We use `pytest <https://pytest.org>`_,
```

Comment:
my preference would be that we standardize around `pytest` for everything.  Any objections, @geowurster @perrygeo @sgillies ?